dbms/src/Common/SensitiveDataMasker.cpp

```diff
@@ -61,15 +64,47 @@ class SensitiveDataMasker::MaskingRule
     uint64_t apply(std::string & data) const
     {
         auto m = RE2::GlobalReplace(&data, regexp, replacement);
+#ifndef NDEBUG
         matches_count += m;
+#endif
         return m;
     }
 
     const std::string & getName() const { return name; }
     const std::string & getReplacementString() const { return replacement_string; }
+#ifndef NDEBUG
     uint64_t getMatchesCount() const { return matches_count; }
+#endif
+
 };
 
+bool SensitiveDataMasker::sensitive_data_masker_initialized = false;
+std::unique_ptr<SensitiveDataMasker> SensitiveDataMasker::sensitive_data_masker = nullptr;
+std::mutex SensitiveDataMasker::mutex;
```

Comment:
I only see one usage of this mutex in set(). What are the concurrent modifications it protects against? (As usual, better put in comment so that further readers can see it straight from the code).

Reply:
Just in case if someday two concurrent attempts to initialize instance will happen.

Reply:
If we need any code in the future, we will add it then. Now, we should understand what is the current situation, write the code that is relevant to it.

If we're sure that the set() is used from a single thread, and not used concurrently with get(), we should not have any synchronization there. We may mention in the comment that the methods have no synchronization, and explain how they are used and why they don't need any.

Or if we do need synchronization, we should also explain why -- when is set() called twice, or when is it called concurrently with get(). As we discussed on Telegram, I hope this is not the case, because the masker is initialized on startup when there are no threads yet, and lives until the server process is destroyed.